### PR TITLE
Fix: handle API Key error in reducer

### PR DESCRIPTION
**Pull request by seanyellowd**
proper redux state fixes the bug.
https://jira.openware.work/browse/EMI-320

### CODE REVIEW

File: src/modules/user/apiKeys/reducer.ts [hunk 1]
```diff
@@ -42,13 +44,15 @@ export const apiKeysReducer = (state = initialApiKeysState, action: ApiKeysActio
                 ...state,
                 apiKeys: sliceArray(action.payload.apiKeys, defaultStorageLimit()),
                 dataLoaded: true,
+                error: undefined,
```
Comment (josadcha) [inline src/modules/user/apiKeys/reducer.ts:47]: add please also dataLoaded: true, for each successful action

**Review by josadcha — CHANGES_REQUESTED** [review]

File: src/modules/user/apiKeys/reducer.ts (hunk 1 above)
Comment (seanyellowd) [reply src/modules/user/apiKeys/reducer.ts:49]: dataLoaded for fetching,
added success value for create, delete, update

**Review by josadcha — APPROVED** [review]

File: src/modules/user/apiKeys/reducer.ts [hunk 2]
```diff
@@ -23,13 +25,15 @@ export interface ApiKeysState {
...
     nextPageExists: boolean;
+    error?: CommonError;
 }
 
 export const initialApiKeysState: ApiKeysState = {
     apiKeys: [],
     dataLoaded: false,
     modal: {
         active: false,
+        success: false,
```
Comment (akhlopiachyi) [inline src/modules/user/apiKeys/reducer.ts:36]: Why did you add this 'success' property?

File: src/modules/user/apiKeys/reducer.ts [hunk 3]
```diff
@@ -42,13 +46,16 @@ export const apiKeysReducer = (state = initialApiKeysState, action: ApiKeysActio
...
+                error: undefined,
                 pageIndex: action.payload.pageIndex,
                 nextPageExists: action.payload.nextPageExists,
             };
         case API_KEY_CREATE:
             return {
                 ...state,
                 apiKeys: state.apiKeys.concat(action.payload),
+                modal: {...state.modal, success: true},
+                error: undefined,
```
Comment (akhlopiachyi) [inline src/modules/user/apiKeys/reducer.ts:58]: You can reset error value in the FETCH actions
You shouldn't set undefined to it if it's already undefined

File: src/modules/user/apiKeys/reducer.ts (hunk 2 above)
Comment (seanyellowd) [reply src/modules/user/apiKeys/reducer.ts:36]: now I see it's not necessary. I will remove.

**Comment by josadcha** [comment]
so please, add dataLoaded for actions as I asked you. If post actions are successful redux state should be updated. In ApiKeys container there is a check if data has been loaded

**Review by akhlopiachyi — APPROVED** [review]
